src/Plugin/Action/IndexNodeInSearchApi.php

```diff
@@ -0,0 +1,122 @@
+use Drupal\Core\Extension\ModuleHandler;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Indexes a node in Search API.
+ *
+ * @Action(
+ *   id = "index_node_in_search_api",
+ *   label = @Translation("Index a node in Search API"),
+ *   type = "node"
+ * )
+ */
+class IndexNodeInSearchApi extends ConfigurableActionBase implements ContainerFactoryPluginInterface {
+  use StringTranslationTrait;
+
+  /**
+   * Module handler.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandler
+   */
+  protected ModuleHandler $moduleHandler;
+
+  /**
+   * Constructor.
+   *
+   * @param array $configuration
+   *   The plugin configuration, i.e. an array with configuration values keyed
+   *   by configuration option name. The special key 'context' may be used to
+   *   initialize the defined contexts by setting it to an array of context
+   *   values keyed by context names.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Extension\ModuleHandler $moduleHandler
+   *   The Module Handler.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ModuleHandler $moduleHandler) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->moduleHandler = $moduleHandler;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('module_handler')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form['index'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Index'),
+      '#default_value' => $this->configuration['index'],
+      '#required' => TRUE,
+    ];
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    $this->configuration['index'] = $form_state->getValue('index');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access($object, AccountInterface $account = NULL, $return_as_object = FALSE) {
+    $result = AccessResult::allowed();
+    return $return_as_object ? $result : $result->isAllowed();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute($node = NULL) {
+    if (!$node) {
+      return;
+    }
+    if (!$this->moduleHandler->moduleExists('search_api')) {
+      return;
+    }
+
+    // Get the Search API index to update.
+    // phpcs:disable
```

Comment:
Any suggestions to avoid/improve using `phpcs:disable`? We can't put a `use` statement without requiring the search_api module; which I really don't want to add.